sdks/python/apache_beam/runners/interactive/dataproc/dataproc_cluster_manager.py

```diff
@@ -0,0 +1,154 @@
+  required for creating and deleting Dataproc clusters for use
+  under Interactive Beam.
+  """
+  DEFAULT_NAME = 'interactive-beam-cluster'
+
+  def __init__(self, project_id=None, region=None, cluster_name=None):
+    """Initializes the DataprocClusterManager with properties required
+    to interface with the Dataproc ClusterControllerClient.
+    """
+
+    self.project_id = project_id
+    if region == 'global':
+      # The global region is unsupported as it will be eventually deprecated.
+      raise ValueError('Clusters in the global region are not supported.')
+    elif region:
+      self.region = region
+    else:
+      _LOGGER.warning(
+          'No region information was detected, defaulting Dataproc cluster '
+          'region to: us-central1.')
+      self.region = 'us-central1'
+
+    if cluster_name:
+      _LOGGER.warning(
+          'A user-specified cluster_name has been detected. '
+          'Please note that you will have to manually delete the Dataproc '
+          'cluster that will be created under the name: %s',
+          cluster_name)
+      self.cluster_name = cluster_name
+    else:
+      self.cluster_name = self.DEFAULT_NAME
+
+    self.cluster_client = dataproc_v1.ClusterControllerClient(
+        client_options={
+            'api_endpoint': f'{self.region}-dataproc.googleapis.com:443'
+        })
+
+  def _create_cluster(self, cluster):
+    """Attempts to create a cluster using attributes that were
+    initialized with the DataprocClusterManager instance.
+
+    Args:
+      cluster: Dictionary representing Dataproc cluster
+    """
+    try:
+      self.cluster_client.create_cluster(
+          request={
+              'project_id': self.project_id,
+              'region': self.region,
+              'cluster': cluster
+          })
+      _LOGGER.info('Cluster created successfully: %s', self.cluster_name)
+    except Exception as e:
+      if e.code == 409:
+        if self.cluster_name == self.DEFAULT_NAME:
+          _LOGGER.info(
+              'Cluster %s already exists. Continuing...', self.DEFAULT_NAME)
+        else:
+          _LOGGER.error(
+              'Cluster already exists - unable to create cluster: %s',
+              self.cluster_name)
+          raise ValueError(
+              'Cluster {} already exists!'.format(self.cluster_name))
+      elif e.code == 403:
+        _LOGGER.error(
+            'Due to insufficient project permissions, '
+            'unable to create cluster: %s',
+            self.cluster_name)
+        raise ValueError(
+            'You cannot create a cluster in project: {}'.format(
+                self.project_id))
+      elif e.code == 501:
+        _LOGGER.error('Invalid region provided: %s', self.region)
+        raise ValueError('Region {} does not exist!'.format(self.region))
+      else:
+        _LOGGER.error('Unable to create cluster: %s', self.cluster_name)
+        raise e
+
+  # TODO(victorhc): Add support for user-specified pip packages
```

Comment:
It seems that the Python packages are installed in many different ways based on the Dataproc image version: https://cloud.google.com/dataproc/docs/tutorials/python-configuration.

- We may pick a stable version for now: `image-version` (I assume it's 1.5 and Python 3.7 happens to be the version we use in notebooks) and install `apache-beam[gcp]==${kernel_version}`
- We still need `metadata` support for `flink-start-yarn-session=true`
- We might need [Initialization actions](https://cloud.google.com/dataproc/docs/concepts/configuring-clusters/init-actions) for more complicated setups.

These are TODO items we can leave for the next PRs.

Reply:
Sounds good! I've jotted these items down to be addressed in future work.